Modules/_elementtree.c

```diff
@@ -230,6 +230,8 @@ typedef struct {
 
     PyObject* _children[STATIC_CHILDREN];
 
+    /* incremented whenever the object is externally mutated */
+    size_t version;
```

Comment:
question: won't just `bool locked;` be enough? 

if some object is locked, we raise an error if any mutation happens. we lock the object when we call python code from C code.